### DISCUSSION TITLE
Issue with recursive iteration and arrays

### DISCUSSION THREAD

**Discussion by CJCombrink** [nlohmann/json#4318] [Q&A]
I am trying to recursively iterate a JSON Object but seem to have issues with an array.

I have code sort of based on this code on Stack Overflow: https://stackoverflow.com/a/73323420/991000
I started with the code posted on the same question from @nlohmann but I need the full paths (`json_pointers`) to the elements.

The issue is that inside the  recursive function when I call `it.key` I get the following runtime exception:
```
terminate called after throwing an instance of 'nlohmann::json_v3_11_1::detail::invalid_iterator'
  what():  [json.exception.invalid_iterator.207] cannot use key() for non-object iterators
Program terminated with signal: SIGSEGV
```

From my understanding from the [docs](https://json.nlohmann.me/features/iterators/#iterators) the `it.key()` function is supposed to give me the array index, from the note:
> When iterating over an array, `key()` will return the index of the element as string. For primitive types (e.g., numbers), `key()` returns an empty string.

See my example on compiler explorer: https://godbolt.org/z/Yr8jdhqoc

Am I doing anything wrong or how would I get this to work as expected. 
Or is this a bug?

I am using `nlohmann_json/3.11.2`

**Comment by nomadwolf0** [comment]
It looks like only the iterators into the collection returned by `basic_json::items()` supports that use of `it.key()`.  The regular iterators of `basic_json` do not have this feature.  Not sure if that's intentional, but that's what seems to be currently implemented.

**Comment by nomadwolf0** [comment]
Slightly uglier than what I had hoped, but here's a working version iterating over `items()`: [https://godbolt.org/z/vfvWq6e4x](https://godbolt.org/z/vfvWq6e4x)
